databricks/koalas/generic.py

```diff
@@ -311,6 +312,59 @@ def abs(self):
         # TODO: The first example above should not have "Name: abs(0)".
         return _spark_col_apply(self, F.abs)
 
+    def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None) \
```

Comment:
does the type hint prevent one from passing in an array_like value at runtime? If not, perhaps we should add a check in the code to throw unsupported exception for array_like values.

Reply:
No, the type hint only warns in IDEs like PyCharm or when using mypy. So I've added a check with d5ea5b9 that throws an exception when list-like values (which include array-like values) are passed as 'lower' or 'upper'.

Reply:
Got it. Thanks.

is there a way in the type hint framework to indicate if the class is a DataFrame return a DataFrame; otherwise return a Series? If there isn't, I'd probably prefer just defining clip twice, once in DataFrame and once in Series.

The Series one can just delegate to the DataFrame one and take the first column. They can also share the same doc by just doing `clip.__doc__ = DataFrame.clip.__doc__`.

Reply:
@floscha not sure if you see this comment. Let me know what you think.

Reply:
I've seen your comment but I was thinking about a good way to use type hints with the *generic* module. However, I ended up splitting the method over the *frame* and *series* modules with d07c0a5 like you've suggested. This resulted in one line for the *series*  module where the `clip` method from *frame* is called. After all, I believe this implementation is much cleaner than the one before (no type checks everywhere), especially when dealing with mypy.

Regarding the docs, I decided to have them separate which I think works nicer when also having separated methods.